Source/WebCore/rendering/RenderElement.h

```diff
@@ -521,37 +524,37 @@ inline bool RenderElement::shouldApplySizeOrStyleContainment(bool containsAccord
 
 inline bool RenderElement::shouldApplyLayoutContainment() const
 {
-    return shouldApplyLayoutOrPaintContainment(style().containsLayout());
+    return shouldApplyLayoutOrPaintContainment(style().containsLayout() || style().contentVisibility() != ContentVisibility::Visible);
```

Comment:
Maybe this test could be in containsLayout() function? Same for containsPaint() etc.

Reply:
I prefer to keep this as-is for now. Future work on this (forced layout/content-visibility: auto) may need to change/extend these code sections and it is not quite clear yet how.